src/pages/ProjectSettings.tsx

```diff
@@ -110,10 +112,18 @@ export const ProjectSettings: React.FC = () => {
     ? projectType === ProjectType.Project || repoData
     : repoData;
```

Comment:
I know this isn't related but this ternary is confusing to me. Couldn't you just check? 

```suggestion
    const hasData = projectData || repoData
```

Reply:
Not exactly, if the project type is an attached project we need to ensure that there is both project and repo data, not one or the other. I agree it could be improved though now that we are have a `ProjectType` defined. Something like
```js
 const hasData = projectType === ProjectType.AttachedProject
     ? projectData && repoData
     : projectData || repoData;
    ? projectType === ProjectType.Project || repoData
    : repoData;
```
🤷‍♀️

Reply:
Thanks for clarifying i wasn't sure how important projectType was. 
Could we use @sophstad  suggestion with a slight change so that hasData is a boolean and not an object.

```js
 const hasData = projectType === ProjectType.AttachedProject
   ? !!projectData && !!repoData
   : !!projectData || !!repoData;
```
```js
let projectData = { "some": "object"}
undefined
let repoData = { "other": "object" }
undefined
let hasData = true ? projectData && repoData : projectData || repoData;
undefined
hasData
{other: 'object'}
let hasData = false ? projectData && repoData : projectData || repoData;
undefined
hasData
{some: 'object'}
hasData = false ? !!projectData && !!repoData : !!projectData || !!repoData;
true
hasData
true
hasData = true ? !!projectData && !!repoData : !!projectData || !!repoData;
true
projectData = undefined
undefined
hasData = true ? !!projectData && !!repoData : !!projectData || !!repoData;
false
hasData = false ? !!projectData && !!repoData : !!projectData || !!repoData;
true
repoData = undefined
undefined
hasData = false ? !!projectData && !!repoData : !!projectData || !!repoData;
false
```

Reply:
Can you please create a ticket for it? This is not related to this PR so I'd prefer to keep the work separate.

Reply:
Done :) https://jira.mongodb.org/browse/EVG-16515